core/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java

```diff
@@ -0,0 +1,313 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import javax.annotation.concurrent.Immutable;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.DateUtil;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Row.MissingCellPolicy;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import tech.tablesaw.api.ColumnType;
+import tech.tablesaw.api.DoubleColumn;
+import tech.tablesaw.api.IntColumn;
+import tech.tablesaw.api.LongColumn;
+import tech.tablesaw.api.Table;
+import tech.tablesaw.columns.Column;
+
+@Immutable
+public class XlsxReader {
+
+    public List<Table> read(XlsxReadOptions options) throws IOException {
+        byte[] bytes = null;
+        InputStream input = getInputStream(options, bytes);
+        List<Table> tables = new ArrayList<Table>();
+        try (XSSFWorkbook workbook = new XSSFWorkbook(input)) {
+            for (Sheet sheet : workbook) {
+                int[] tableArea = findTableArea(sheet);
+                if (tableArea != null) {
+                    Table table = createTable(sheet, tableArea, options);
+                    tables.add(table);
+                }
+            }
+            return tables;
+        } finally {
+            if (options.reader() == null) {
+                // if we get a reader back from options it means the client opened it, so let
+                // the client close it
+                // if it's null, we close it here.
+                input.close();
+            }
+        }
+    }
+
+    private Boolean isBlank(Cell cell) {
+        switch (cell.getCellType()) {
+        case STRING:
+            if (cell.getRichStringCellValue().length() > 0) {
+                return false;
+            }
+            break;
+        case NUMERIC:
+            if (DateUtil.isCellDateFormatted(cell) ? cell.getDateCellValue() != null
+                    : cell.getNumericCellValue() != 0) {
+                return false;
+            }
+            break;
+        case BOOLEAN:
+            if (cell.getBooleanCellValue()) {
+                return false;
+            }
+            break;
+        case BLANK:
+            return true;
+        default:
+            break;
+        }
+        return null;
+    }
+
+    private ColumnType getColumnType(Cell cell) {
+        switch (cell.getCellType()) {
+        case STRING:
+            return ColumnType.STRING;
+        case NUMERIC:
+            return DateUtil.isCellDateFormatted(cell) ? ColumnType.LOCAL_DATE_TIME : ColumnType.SHORT;
+        case BOOLEAN:
+            return ColumnType.BOOLEAN;
+        default:
+            break;
+        }
+        return null;
+    }
+
+    private int[] findTableArea(Sheet sheet) {
```

Comment:
Could you add a Javadoc describing what the return value is? Or better yet would be to use a static inner class instead of an array since that would be self-documenting